src/messenger_packets.md

```diff
@@ -0,0 +1,147 @@
+Title: Packets of messenger
+
+This is a description of packets of messenger layer of Tox.
+
+## FILE_DATA
+
+FileData packet is used to transfer sender's data file to a friend.
+It holds `file_id` which is one byte long, means that a tox client can send maximum 256 files concurrently.
+Also it means that two friends can send 512 files to each other concurrently.
+
+Serialized form:
+
+Length    | Content
+--------- | ------
+`1`       | `0x52`
+`1`       | `file_id`
+`0..1371` | file data piece
+
+One FileData packet can hold maximum 1371 bytes.
+To send whole file which is bigger than 1371 bytes, messenger's file sending function need to be called many times.
+Multiple calling file sending function of messenger must be done by the user of tox-rs, Tox-rs provides only an api for sending a chunk of file data.
+Also, assembling chunks of file data received by tox-rs must be done by the user of tox-rs.
+File sending module always checks the length of sent bytes by length of file to send.
+When the sent bytes exceed the length of file to send, it must stop sending file data.
+
+## FILE_CONTROL
+
+This packet is used to control transferring sender's file to a friend.
+If a peer of connection wants to pause, kill, seek or accept transferring file, it must use this packet.
+
+Serialized form:
+
+Length    | Content
+--------- | ------
+`1`       | `0x51`
+`1`       | Whether it is sending or receiving, 0 = sender, 1 = receiver
```

Comment:
I don't understand how it works. Do we really need to send that byte in the serialized form? It makes no sense to me:

Assume a friend sends to us: `[0x52, 0, ....]` (0 because he is the sender), we get it and we see that we are the sender of the packet. What's that for?

Reply:
https://github.com/TokTok/c-toxcore/blob/0aad180d1e1d7dae7cbf2868e139af0a4dc34fe7/toxcore/Messenger.c#L1684

When we got FILE_CONTROL packet, we find which file we are operating(sending or receiving).
Because `file_id` is one bytes, we can send maximum 256 files and receive maximum 256 files concurrently.
So, we need `SendDirection` or `send_receive` flag combined with `file_id` to find which file we want to control.

Reply:
Thanks.